src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/CherryPickOperation.cs

```diff
@@ -0,0 +1,201 @@
+    private readonly IVmrPatchHandler _patchHandler;
+    private readonly IVersionDetailsParser _versionDetailsParser;
+    private readonly ILocalGitRepoFactory _localGitRepoFactory;
+    private readonly IVmrInfo _vmrInfo;
+    private readonly ILogger<CherryPickOperation> _logger;
+
+    public CherryPickOperation(
+        CherryPickCommandLineOptions options,
+        IProcessManager processManager,
+        IFileSystem fileSystem,
+        IVmrPatchHandler patchHandler,
+        IVersionDetailsParser versionDetailsParser,
+        ILocalGitRepoFactory localGitRepoFactory,
+        IVmrInfo vmrInfo,
+        ILogger<CherryPickOperation> logger)
+    {
+        _options = options;
+        _processManager = processManager;
+        _fileSystem = fileSystem;
+        _patchHandler = patchHandler;
+        _versionDetailsParser = versionDetailsParser;
+        _localGitRepoFactory = localGitRepoFactory;
+        _vmrInfo = vmrInfo;
+        _logger = logger;
+    }
+
+    public override async Task<int> ExecuteAsync()
+    {
+        try
+        {
+            return await ExecuteInternalAsync();
+        }
+        catch (Exception e)
+        {
+            _logger.LogError("Cherry-pick operation failed: {message}", e.Message);
+            _logger.LogDebug("{exception}", e);
+            return Constants.ErrorCode;
+        }
+    }
+
+    private async Task<int> ExecuteInternalAsync()
+    {
+        // Step 1: Determine if we're in VMR or repo based on source-manifest.json existence
+        var currentDirectory = Environment.CurrentDirectory;
+        var gitRoot = _processManager.FindGitRoot(currentDirectory);
+        var sourceManifestPath = Path.Combine(gitRoot, VmrInfo.DefaultRelativeSourceManifestPath.Path);
+        var isInVmr = _fileSystem.FileExists(sourceManifestPath);
+
+        _logger.LogInformation("Cherry-pick operation starting from {location}", isInVmr ? "VMR" : "repository");
+
+        // Step 2: Get mapping name from Version.Details.xml
+        string mappingName;
+        if (isInVmr)
+        {
+            // When in VMR, we need to get mapping from the source repo's Version.Details.xml
+            mappingName = GetMappingFromRepo(_options.SourceRepo);
+        }
+        else
+        {
+            // When in repo, get mapping from current repo's Version.Details.xml
+            mappingName = GetMappingFromRepo(gitRoot);
+        }
+
+        _logger.LogInformation("Detected mapping name: {mappingName}", mappingName);
+
+        if (isInVmr)
+        {
+            // Cherry-pick from VMR to repo
+            return await CherryPickFromVmrToRepoAsync(gitRoot, mappingName);
+        }
+        else
+        {
+            // Cherry-pick from repo to VMR
+            return await CherryPickFromRepoToVmrAsync(gitRoot, mappingName);
+        }
+    }
+
+    private string GetMappingFromRepo(string repoPath)
+    {
+        var versionDetailsPath = Path.Combine(repoPath, "eng", "Version.Details.xml");
```

Comment:
There' a class `VersionFiles` somewhere in the repo which has the path to the `Version.Details.xml` defined. Use it from there.